python/lsst/meas/extensions/scarlet/multiBandDetectionTask.py

```diff
@@ -0,0 +1,268 @@
+        """Add schema keys to the input schema
+
+        Currently there are no keys added, but that could change in the future
+        so we'll keep this method around.
+        """
+        pass
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+        inputs["filters"] = [dRef.dataId["band"] for dRef in inputRefs.coadds]
+        inputs["idFactory"] = self.config.idGenerator.apply(
+            butlerQC.quantum.dataId
+        ).make_table_id_factory()
+        outputs = self.run(**inputs)
+        butlerQC.put(outputs, outputRefs)
+
+    def run(
+        self,
+        coadds: list[afwImage.Exposure],
+        bands: list[str],
+        idFactory: IdFactory,
+        peaks: PeakCatalog | None = None,
+    ) -> pipeBase.Struct:
+        if (not self.config.doDetectPeaks) and (peaks is None):
+            raise ValueError("'peaks' must be provided if doDetectPeaks is False")
+        # Create a MulitbandExposure from the list of coadds
+        mCoadd = afwImage.MultibandExposure.fromExposures(bands, coadds)
+        xmin, ymin = mCoadd.getBBox().getMin()
+        # Detect footprints in the detection image
+        footprints = scl.detect.detect_footprints(
+            images=mCoadd.image.array,
+            variance=mCoadd.variance.array,
+            origin=(ymin, xmin),
+            min_separation=self.config.minPeakDistance,
+            min_area=self.config.minFootprintArea,
+            peak_thresh=self.config.minPeakSNR,
+            footprint_thresh=self.config.minFootprintSNR,
+            find_peaks=False,
+            remove_high_freq=False,
+            min_pixel_detect=self.config.minPixelDetect,
+        )
+
+        # Create a source catalog from the detected footprints
+        table = SourceTable.make(self.schema, idFactory)
+        sources = SourceCatalog(table)
+        for footprint in footprints:
+            fp_ymin, fp_xmin = footprint.bbox.origin
+            spans, _ = getSpanSetFromImages(
+                footprint.data, xy0=Point2I((xmin + fp_xmin, ymin + fp_ymin))
+            )
+            _footprint = AfwFootprint(spans)
+            source = sources.addNew()
+            source.setFootprint(_footprint)
+
+        if self.config.doDetectPeaks:
+            # Detect peaks in the detection image
+            # and update the footprints in place.
+            peaks = self._detectPeaks(mCoadd, footprints)
+
+        # Attach peaks from an external catalog to the footprints.
+        self._attachExternalPeaks(peaks, footprints, sources, mCoadd.getBBox())
+
+        return pipeBase.Struct(footprints=footprints, sources=sources)
+
+    def _detectPeaks(
+        self,
+        mCoadd: afwImage.MultibandExposure,
+        footprints: Sequence[scl.detect.Footprint],
+    ):
+        """Detect peaks in the detection image"""
+        # Build the noise weighted detection image.
+        sigma = np.median(np.sqrt(mCoadd.variance.array), axis=(1, 2)) / 2
```

Comment:
Why half of the median? Should this be configurable?

Reply:
I think I was testing something and just forgot to remove that 2. This is a good catch, because it wouldn't make any difference anyway.